api/envoy/extensions/filters/http/rate_limit_quota/v3/rate_limit_quota.proto

```diff
@@ -0,0 +1,154 @@
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#not-implemented-hide:]
+// [#protodoc-title: Rate Limit Quota]
+// Rate limit :ref:`configuration overview <config_http_filters_rate_limit_quota>`.
+// [#extension: envoy.filters.http.rate_limit_quota]
+
+message RateLimitQuotaFilterConfig {
+  // [#comment:TODO(sergiitk): preview mode matchers]
+
+  // Configuration for the gRPC service that hosts an external rate limit
+  // provider. If not specified, any calls to the rate limit service will
+  // immediately return success.
+  config.core.v3.ConfigSource rlqs_server = 1;
+
+  // The application domain to use when calling the service.
+  string domain = 2;
+
+  // Unified matchers.
+  xds.type.matcher.v3.Matcher bucket_matchers = 3;
+}
+
+// [#next-free-field: 6]
+message RateLimitQuotaBucketSettings {
+  // Configures the behavior when the first request has been matched to
+  // the bucket, and the data plane is waiting for the RLQS to return the
+  // first quota assignment.
+  message NoAssignmentBehavior {
+    oneof behavior {
+      option (validate.required) = true;
+
+      // Allow or Deny the requests until the bucket is assigned a quota.
+      // If unset, allow all.
+      type.v3.RateLimitStrategy.BlanketRule blanket_rule = 1;
+
+      // Queue the requests until the bucket is assigned a quota.
+      // The data plane maintains a separate queue for each bucket that has
+      // pending requests.
+      // When the server returns quota assignment, the data plane begins
+      // dequeuing the requests in the FIFO order, and processing them according
+      // to the assignment. No individual request/response matching needed.
+      //
+      // .. note::
+      // To avoid memory exhaust, queue configuration allows to
+      // to impose constraints on the maximum number of pending requests
+      // in the queue, and choose what to do with the request over the limit.
+      type.v3.RateLimitStrategyQueue queue_requests = 2;
+    }
+  }
+
+  // Specifies the behavior when the bucket's assignment has expired, and
+  // hasn't been refreshed due to server's unavailability.
+  message ExpiredAssignmentBehavior {
+    message FallbackLastAssignment {
+      // Limit the time the last assignment is used as a fallback.
+      // [#comment:TODO(sergiitk): what to do after that?]
+      google.protobuf.Duration time_to_live = 1;
+    }
+
+    oneof behavior {
+      option (validate.required) = true;
+
+      // Pre-configured local rate limit strategy.
+      type.v3.RateLimitStrategy fallback_rate_limit = 1;
+
+      // Reuse last quota assignment associated to the bucket.
+      FallbackLastAssignment fallback_last_assignment = 2;
+    }
+  }
+
+  message DenyResponseSettings {
+    // Status code to use to fail requests when quota is exceeded.
+    // [#comment:TODO: figure out the right defaults for these fields!]
+    int32 http_status_code = 1;
```

Comment:
I think Envoy prefers using wrapper types when defaults do not match 0.

Reply:
Thanks for the advice! I ended up replacing this one with `type.v3.HttpStatus` (see https://github.com/envoyproxy/envoy/pull/19793/commits/8776b2b23fcd8c29eddbb78861e459ef54d251c3), but I keep your advice in mind for other places.